R/dust_generator.R

```diff
@@ -250,6 +243,35 @@ dust_generator <- R6::R6Class(
     set_pars = function(pars) {
     },
 
+    ##' @description Update one or more components of the model state.
+    ##'   This method can be used to update any or all of `pars`, `state` and
+    ##'   `step`.  If `pars` is given and `state` is not, then by default we
+    ##'   will update the model internal state according to your model's
+    ##'   initial conditions - use `set_initial_state = FALSE` to prevent
+    ##'   this.
```

Comment:
I might expect the opposite behaviour as default, but I presume this way is how this is being used by sircovid etc

Reply:
Indeed, that's the trick. We could change this to be if we have pars AND step NOT state, perhaps, which is the least surprising set and I think fully compatible with sircovid (this was just the old `reset` method)

Reply:
Updated to be less enthusiastic about setting initial conditions